### ISSUE TITLE
[Suggestion/Discussion] Block readability and the new player experience

### ISSUE THREAD

**Issue by Womble-TC** [tsunamayo/Starship-EVO#4940]
### Long Description of the issue:

Apologies in advance, this will be quite long, and I could be completely wrong. However, something I’ve come across while trying to use the new mining/crafting system is how confusing it is. For new players, I can only imagine how difficult it would be to use without any kind of in-depth tutorial.

While the assets are probably not final, one thing I noticed the new machines are missing is any kind of uniform design language. 

Existing systems in the game (mainly the weapons and generators) have shared features to show what can be connected to them and where:

![image](https://user-images.githubusercontent.com/29961381/192178523-f17e43fb-712a-419c-a60f-2576ff7c873e.png)
![image](https://user-images.githubusercontent.com/29961381/192178539-bd12344c-1628-4abf-9264-57e8d4045f39.png)

This is great! because without checking the wiki or playing a systems tutorial, its quite obvious that a capacitor can be used on both the shield generator and the reactor, as the texture and general design is shared. Likewise, both the beam and the laser weapons have clear nodes to which different modules can be attached. 

Its also satisfying to use systems, as there’s nice visual and audio feedback when compatible systems are joined together, also further solidifying the idea to new players that these parts should go together.

The issue I have with the _current_ implementation of the furnace and assembler is that there’s no clear indication of how these blocks should be used:

![image](https://user-images.githubusercontent.com/29961381/192179329-189956a7-6d4a-4004-9dbd-32827c830933.png)
![image](https://user-images.githubusercontent.com/29961381/192179365-c7c2fcfc-b4ac-40d9-b61a-5df30872c2b7.png)

An example of rather basic but clear visual design can be seen in space engineers. While the system blocks themselves can be very complex, its very clear where tubes should be connected, and the fact that the system requires a tube connection:

![image](https://user-images.githubusercontent.com/29961381/192179812-c33a1e27-7cb3-4a66-852c-49eeb92df730.png)
![image](https://user-images.githubusercontent.com/29961381/192179761-553d48a0-022d-47b3-a6aa-7c0dd0b37699.png)

Since we have the magic of stretch-tech in Starship Evo, it wouldn’t be great to have a fixed connection size. However, a connection ‘zone’ could be at least used. This would act as some kind of recognisable detail used across all systems that require a pipe connection. For this example, I’ve just used the cargo container texture: 

![image](https://user-images.githubusercontent.com/29961381/192180063-83d1f429-74fe-4998-8788-4b5ec897c584.png)

This texture on the side of a system would show a player that tubes can be connected on this texture, and only on this texture. While this _does_ limit the spaces in which tubes can be connected to the system, bring a consistent design language or cargo connection.

Here is an example of how a machine could use such a connection ‘zone’, as well as a mock-up of the furnace if it only used conveyer tubes instead of belts:

![image](https://user-images.githubusercontent.com/29961381/192180291-4534c16d-d130-4eb4-a21f-98fb13f979ae.png)
![image](https://user-images.githubusercontent.com/29961381/192180311-676dcfc5-00e6-4cbc-8540-c6a6bd62fe73.png)

Since the furnace uses belts, it would be ideal in future if this was clearly indicated on the model (I update the current one if needed):

![image](https://user-images.githubusercontent.com/29961381/192180470-89434d9d-b02d-4230-acf7-ef534ccf9ab4.png)

To summarise, I think it’s important that system blocks in future feature some kind of design language. This would hopefully improve the new player experience, and just the overall satisfaction of putting things together in the right place. 

It would also mean that mini tutorials wouldnt be required every time a new system is added, since it would _hopefully_ be visually clear how the new blocks would work with current systems.

Sorry again that this is a lot all at once, it’s just something I feel strongly about. As usual, great work on the game so far!

### Environment
- Starship EVO version: 22w38c

**Comment by tsunamayo** [comment]
@Womble-TC Yeah I agree with that.
To be honest I havent spent any time on this, waiting to see how thing where shaping up as you often need to change plans.

I like your idea of the container sticking out, it is a great tell that you can connect and that is has internal inventory too. I could add some sort of an decal too.
Having a small conveyor on the furnace is a great idea, I think I will do this.
I havent tested the assembler with conveyors, I will fix it. I could swap the assembler texture for a more conveyor like too (I wanted to make something that open/close as the item gets pulled, but that will be too much of hassle.

I am not sure about restricting the connection on the crate though. I wanted to give more crate with a different look too...
Same with pipe, I still want to give the flat tank. So I guess we will have to do with having the player learning that he can connect where he wants.

Cheers
Francois

**Comment by Uncle-Ulty** [comment]
I agree with the idea of a "systematic way to connect mechanisms". As soon a "standard pattern" is implemented, it will be easier to expand it in the future.  Visual indication is a friendly way to teach how the game work, without the need to several windows and explanations

**Comment by Drillz007** [comment]
Agree with womble having things a bit more self explanatory/uniform would be nice

**Comment by COMICLER** [comment]
This perfectly sums up what i feel about the new systems

**Comment by cyberjasse** [comment]
what confused me was how to connect the conveyors from an entity to a child entity. i did not figured out how to do it directly. i finally figured out later.
in the future, without reading wiki, it would be unclear about what can be connected. thats why i agree with this post. But perhaps there is another way to solve the problems
